Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -0,0 +1,442 @@
+                in float aspectRatio, in float maxOrthoSize)
+            {
+                return boundingShape2D != null && 
+                       m_boundingShape2D != null && m_boundingShape2D == boundingShape2D && // same boundingShape?
+                       //!BoundingShapeTransformChanged(boundingShape2D.transform) && // input shape changed?
+                       m_originalPath != null && // first time?
+                       m_confinerStates != null && // cache not empty? 
+                       Mathf.Abs(m_aspectRatio - aspectRatio) < UnityVectorExtensions.Epsilon && // aspect changed?
+                       Mathf.Abs(m_maxOrthoSize - maxOrthoSize) < UnityVectorExtensions.Epsilon; // max ortho changed?
+            }
+
+            private void SetTransformCache(in Transform boundingShapeTransform)
+            {
+                m_boundingShapeScale = boundingShapeTransform.localScale;
+                m_boundingShapeRotation = boundingShapeTransform.rotation;
+            }
+
+            private void SetLocalToWorldDelta()
+            {
+                Transform boundingShapeTransform = m_boundingShape2D.transform;
+                m_localToWorldDelta.position = boundingShapeTransform.position;
+                m_localToWorldDelta.rotation = Quaternion.Inverse(m_boundingShapeRotation) * 
+                                               boundingShapeTransform.rotation;
+                
+                Vector3 localScale = boundingShapeTransform.localScale;
+                localScale.x = Math.Abs(m_boundingShapeScale.x) < UnityVectorExtensions.Epsilon
+                    ? 0
+                    : localScale.x / m_boundingShapeScale.x;
+                localScale.y = Math.Abs(m_boundingShapeScale.y) < UnityVectorExtensions.Epsilon
+                    ? 0
+                    : localScale.y / m_boundingShapeScale.y;
+                localScale.z = Math.Abs(m_boundingShapeScale.z) < UnityVectorExtensions.Epsilon
+                    ? 0
+                    : localScale.z / m_boundingShapeScale.z;
+                m_localToWorldDelta.localScale = localScale;
+            }
+        }
+        private ShapeCache m_shapeCache;
+
+        private GameObject localToWorldTransformHolder;
+        private void Start()
+        {
+            if (m_shapeCache.m_localToWorldDelta == null)
+            {
+                localToWorldTransformHolder = new GameObject("LocalToWorldTransform");
+                localToWorldTransformHolder.hideFlags = HideFlags.HideAndDontSave;
+                m_shapeCache.m_localToWorldDelta = localToWorldTransformHolder.transform;
+            }
+        }
+        
+        private void OnValidate()
+        {
+            m_Damping = Mathf.Max(0, m_Damping);
+            m_MaxOrthoSize = Mathf.Max(0, m_MaxOrthoSize);
+        }
+
+    #if UNITY_EDITOR
+        internal Color m_gizmoColor = Color.yellow;
+        private void OnDrawGizmos()
```

Comment:
Gizmo drawing should not happen in runtime module.
Move this code to Confiner2D editor.  Look at CinemachineColliderEditor.cs to see how it's done.